src/java/org/apache/cassandra/service/accord/txn/TxnUpdate.java

```diff
@@ -116,35 +116,38 @@ public ConditionalBlock deserialize(DataInputPlus in) throws IOException
             public void skip(DataInputPlus in) throws IOException
             {
                 in.readUnsignedVInt32();
-                skipWithVIntLength(in);
+                if (in.readBoolean())
+                    skipWithVIntLength(in);
                 SerializePacked.skipPackedSortedIntsAndLength(in);
             }
 
             @Override
             public long serializedSize(ConditionalBlock t)
             {
                 long size = TypeSizes.sizeofUnsignedVInt(t.id);
-                size += serializedSizeWithVIntLength(t.condition.bytes());
+                size += TypeSizes.sizeof(t.condition != null);
```

Comment:
Why is condition `nullable`? I don't believe we assign it anything non-null in real codepaths, and I don't think we *should* either, since we have the `None` condition (which takes up 1 byte rather than zero).